pubsub-service-api.yml

```diff
@@ -476,19 +482,19 @@ components:
           - content
         properties:
           sender:
-            description: the Peer Id of the message's sender
+            description: the peerID of the message's sender
             type: string
             example: "QmcXhKGEAJT9DwwzKeEZsWrYTTGkzBdgeKif2MdAXBswJv"
           content:
-            description: the base64 encoded message's content,
+            description: the base64 encoded message's content
             type: string
             example: "YWJj"
           pubkey:
-            description: the public key associated to the sender's peer ID, formatted as a base64-encoded x509 DER public key
+            description: the public key associated to the sender's peer ID, formatted as a base64-encoded x509 DER public key. If the the signature has not been requested or if the peerID directly embed the public key (for example with ED25519 and secp256k1), this field is omitted.
```

Comment:
Sounds good for small keys, but sending the entire RSA key every time be pretty wasteful, especially if the same peer sends multiple messages: signature will be different, but the Base64 in `pubkey` field will be the same across all messages sent by that peer.

If that becomes a problem, consider is to add a separate `include-pubkey`  for controlling the `pubkey` presence (making it false by default is imo safer default, but also fine for it to be enabled by default, at least you provide users with means of  HIDING  RSA keys and getting them off-band).